pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexOpenRedirectRule.java

```diff
@@ -102,26 +99,12 @@ private void findSafeLiterals(AbstractApexNode<?> node) {
             }
         } else {
             if (node instanceof ASTField) {
-                /*
-                 * sergey.gorbaty: Apex Jorje parser is returning a null from
-                 * Field.getFieldInfo(), but the info is available from an inner
-                 * field. DO NOT attempt to optimize this block without checking
-                 * that Jorje parser actually fixed its bug.
-                 * 
-                 */
-                try {
-                    final Field f = node.getNode().getClass().getDeclaredField("fieldInfo");
-                    f.setAccessible(true);
-                    final StandardFieldInfo fieldInfo = (StandardFieldInfo) f.get(node.getNode());
-                    if (fieldInfo.getType().getApexName().equalsIgnoreCase("String")) {
-                        if (fieldInfo.getValue() != null) {
-                            addVariable(fieldInfo);
-                        }
+                ASTField field = (ASTField) node;
+                if ("String".equalsIgnoreCase(field.getTypeRef())) {
+                    StandardFieldInfo fieldInfo = (StandardFieldInfo) field.getNode().getFieldInfo();
+                    if (fieldInfo.getValue() != null) {
```

Comment:
In my tests, the value of the fieldInfo was always `null`... so, not sure, if this code is even executed.

Reply:
The problem was the quotes: Apex uses single quotes for Strings, not double quotes... with single quotes, there is a value.

Reply:
Is there an option in Jorje to make the parser fail on syntax errors rather than just produce an inconsistent AST? This would help save time if we enable that eg in tests and the designer

Reply:
Good question...
We currently run the compiler with stage "post type resolve": https://github.com/pmd/pmd/blob/f6549d6c0bbc896d9a27838dcc6eac0c5b97ce89/pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/CompilerService.java#L76

There is also a stage called "validate" and "additional validate". Maybe one of these is what we want? I'll try this tomorrow.

Reply:
See #1741